test/e2e/storage/testsuites/snapshottable_stress.go

```diff
@@ -0,0 +1,233 @@
+			framework.Logf("Creating resources for pod %v/%v", i, stressTest.testOptions.NumPods-1)
+
+			volume := CreateVolumeResource(driver, stressTest.config, pattern, t.GetTestSuiteInfo().SupportedSizeRange)
+			stressTest.volumes = append(stressTest.volumes, volume)
+
+			podConfig := e2epod.Config{
+				NS:           f.Namespace.Name,
+				PVCs:         []*v1.PersistentVolumeClaim{volume.Pvc},
+				SeLinuxLabel: e2epv.SELinuxLabel,
+			}
+			pod, err := e2epod.MakeSecPod(&podConfig)
+			framework.ExpectNoError(err)
+			stressTest.pods = append(stressTest.pods, pod)
+
+			if _, err := cs.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{}); err != nil {
+				stressTest.cancel()
+				framework.Failf("Failed to create pod-%v [%+v]. Error: %v", i, pod, err)
+			}
+
+			if err := e2epod.WaitForPodRunningInNamespace(cs, pod); err != nil {
+				stressTest.cancel()
+				framework.Failf("Failed to wait for pod-%v [%+v] turn into running status. Error: %v", i, pod, err)
+			}
+		}
+	}
+
+	cleanup := func() {
+		framework.Logf("Stopping and waiting for all test routines to finish")
+		stressTest.cancel()
+		stressTest.wg.Wait()
+
+		var errs []error
+		for _, snapshot := range stressTest.snapshots {
+			framework.Logf("Deleting snapshot %v", snapshot.Vs.GetName())
+			errs = append(errs, snapshot.CleanupResource())
+		}
+		for _, pod := range stressTest.pods {
+			framework.Logf("Deleting pod %v", pod.Name)
+			errs = append(errs, e2epod.DeletePodWithWait(cs, pod))
+		}
+		for _, volume := range stressTest.volumes {
+			framework.Logf("Deleting volume %v", volume.Pvc.GetName())
+			errs = append(errs, volume.CleanupResource())
+		}
+		errs = append(errs, tryFunc(stressTest.driverCleanup))
+		framework.ExpectNoError(errors.NewAggregate(errs), "While cleaning up resources")
+	}
+
+	ginkgo.BeforeEach(func() {
+		init()
+		createPodsAndVolumes()
+	})
+
+	f.AddAfterEach("cleanup", func(f *framework.Framework, failed bool) {
+		cleanup()
+	})
+
+	ginkgo.It("should support snapshotting of many volumes repeatedly [Slow] [Serial]", func() {
+		// Repeatedly create and delete snapshots of each volume.
+		for i := 0; i < stressTest.testOptions.NumPods; i++ {
+			stressTest.wg.Add(1)
+
+			go func(podIndex int) {
+				defer ginkgo.GinkgoRecover()
+				defer stressTest.wg.Done()
+
+				pod := stressTest.pods[podIndex]
+				volume := stressTest.volumes[podIndex]
+
+				for j := 0; j < stressTest.testOptions.NumSnapshots; j++ {
+					select {
+					case <-stressTest.ctx.Done():
+						return
+					default:
+						framework.Logf("Pod-%v [%v], Iteration %v/%v", podIndex, pod.Name, j, stressTest.testOptions.NumSnapshots-1)
+
+						snapshot := CreateSnapshotResource(snapshottableDriver, stressTest.config, pattern, volume.Pvc.GetName(), volume.Pvc.GetNamespace())
+
+						stressTest.snapshotsMutex.Lock()
+						stressTest.snapshots = append(stressTest.snapshots, snapshot)
+						stressTest.snapshotsMutex.Unlock()
+
```

Comment:
Do we want to randomize the amount of time we wait for a snapshot before cleaning up? That way we can test different states that the snapshot is in when we issue the delete.

Reply:
Do you mean deleting the snapshot while it's still being created? From my understanding, `CreateSnapshotResource()` does not return until the snapshot is available for use, so by the time we call `snapshot.CleanupResource()` it is fully ready.

Reply:
I need to check how we can refactor this to break it into creating the object and then waiting for it to be available.